node/hack/repair_terra/repair.go

```diff
@@ -0,0 +1,369 @@
+			continue
+		}
+		nonceInt, err := strconv.ParseUint(nonce, 10, 32)
+		if err != nil {
+			log.Println("nonce cannot be parsed as int", zap.String("tx_hash", txHash), zap.String("value", blockTime))
+			continue
+		}
+		sequenceInt, err := strconv.ParseUint(sequence, 10, 64)
+		if err != nil {
+			log.Println("sequence cannot be parsed as int", zap.String("tx_hash", txHash), zap.String("value", blockTime))
+			continue
+		}
+		messagePublication := &common.MessagePublication{
+			TxHash:           txHashValue,
+			Timestamp:        time.Unix(blockTimeInt, 0),
+			Nonce:            uint32(nonceInt),
+			Sequence:         sequenceInt,
+			EmitterChain:     vaa.ChainIDTerra,
+			EmitterAddress:   senderAddress,
+			Payload:          payloadValue,
+			ConsistencyLevel: 0, // Instant finality
+		}
+		msgs = append(msgs, messagePublication)
+	}
+	return msgs
+}
+
+// StringToAddress convert string into address
+func StringToAddress(value string) (vaa.Address, error) {
+	var address vaa.Address
+	res, err := hex.DecodeString(value)
+	if err != nil {
+		return address, err
+	}
+	copy(address[:], res)
+	return address, nil
+}
+
+// StringToHash convert string into transaction hash
+func StringToHash(value string) (eth_common.Hash, error) {
+	var hash eth_common.Hash
+	res, err := hex.DecodeString(value)
+	if err != nil {
+		return hash, err
+	}
+	copy(hash[:], res)
+	return hash, nil
+}
+
+func main() {
+	flag.Parse()
+
+	ctx := context.Background()
+
+	missingMessages := make(map[uint64]bool)
+
+	conn, err, admin := getAdminClient(ctx, *adminRPC)
+	defer conn.Close()
+	if err != nil {
+		log.Fatalf("failed to get admin client: %v", err)
+	}
+
+	var TerraEmitter = struct {
+		ChainID vaa.ChainID
+		Emitter string
+	}{vaa.ChainIDTerra, "0000000000000000000000007cf7b764e38a0a5e967972c1df77d432510564e2"}
+
+	log.Printf("Requesting missing messages for %s", TerraEmitter.Emitter)
+
+	msg := nodev1.FindMissingMessagesRequest{
+		EmitterChain:   uint32(vaa.ChainIDTerra),
+		EmitterAddress: TerraEmitter.Emitter,
+		RpcBackfill:    true,
+		BackfillNodes:  common.PublicRPCEndpoints,
+	}
+	resp, err := admin.FindMissingMessages(ctx, &msg)
+	if err != nil {
+		log.Fatalf("failed to run find FindMissingMessages RPC: %v", err)
+	}
+
+	msgs := make([]*db.VAAID, len(resp.MissingMessages))
+	for i, id := range resp.MissingMessages {
+		vId, err := db.VaaIDFromString(id)
+		if err != nil {
+			log.Fatalf("failed to parse VAAID: %v", err)
+		}
+		msgs[i] = vId
+	}
+
+	if len(msgs) == 0 {
+		log.Printf("No missing messages found for %s", TerraEmitter)
+		return
+	}
+
+	lowest := msgs[0].Sequence
+	highest := msgs[len(msgs)-1].Sequence
+
+	log.Printf("Found %d missing messages for %s: %d - %d", len(msgs), TerraEmitter, lowest, highest)
+
+	for _, msg := range msgs {
+		missingMessages[msg.Sequence] = true
+	}
+
+	log.Printf("Starting search for missing sequence numbers...")
+	offset := 0
+	var firstTime bool = true
+	for ((offset > 0) || firstTime) {
+		firstTime = false
```

Comment:
Could you check for offset == 0 instead? Go ints default to zero anyways

Reply:
The offset needs to be 0 the first time through the loop.  The next time it is 0 => no more items to read